compound_status.py

```diff
@@ -292,187 +190,48 @@ class Condensed(Clobberer):
-        status_type = STATUS_NAME_TO_CLASS[self.status]
-        status_msg = self.message
-        return status_type(status_msg)
-
-    def _set(self, status: StatusName, msg: str = ""):
-        """Force-set this status and message.
-
-        Should not be called by user code.
-        """
-        self._user_set = True
-        super()._set(status, msg)
-
-    def unset(self):
-        """Unset all child statuses, as well as any user_set Master status."""
-        super().unset()
-
-        self._user_set = False
-        for child in self.children:
-            child.unset()
-
-    def _snapshot(self) -> _StatusDict:
-        """Serialize Status for storage."""
-        dct = super()._snapshot()
-        dct["type"] = "master"
-        dct["user_set"] = self._user_set
-        return dct
-
-    def _restore(self, dct: _StatusDict):
-        """Restore Status from stored state."""
-        type_ = dct.get("type", None)
-        assert type_, type_  # type guard
-        assert type_ == "master", type_
-
-        status = dct.get("status", None)
-        message = dct.get("message", None)
-        user_set = dct.get("user_set", None)
-
-        assert status is not None, status
-        assert message is not None, message
-        assert user_set is not None, user_set
-
-        self._status = status
-        self._message = message
-        self._user_set = user_set
-
-    def __repr__(self):
-        if not self.children:
-            return "<MasterStatus -- empty>"
-        if self.status == "unknown":
-            return "unknown"
-        return str(self.coalesce())
+        if skip_unknown and status == "unknown":
+            continue
+        msgs.append("{0} {1}".format(count, status))
+    return "; ".join(msgs)
 
 
 class StatusPool(Object):
     """Represents the pool of statuses available to an Object."""
 
-    # whether unknown statuses should be omitted from the master message
-    SKIP_UNKNOWN = False
-    # whether the status should be committed automatically when the hook exits
-    AUTO_COMMIT = True
-    # key used to register handle
-    KEY = "status_pool"
-
-    if TYPE_CHECKING:
-        _statuses = {}  # type: Dict[str, Status]
-        _charm: CharmBase
-        master = MasterStatus()  # type: MasterStatus
-        _manual_priorities = False  # type: bool
-        _priority_counter = 0  # type: int
-
-    def __init__(self, charm: CharmBase, key: Optional[str] = None):
-        super().__init__(charm, key or self.KEY)
-        # skip setattr
-        self.__dict__["master"] = MasterStatus()
-        self.__dict__["_statuses"] = {}
-        self.__dict__["_manual_priorities"] = False
-        self.__dict__["_priority_counter"] = 0
-
-        stored_handle = Handle(self, StoredStateData.handle_kind, "_status_pool_state")
+    def __init__(
+        self,
+        charm: CharmBase,
+        key: str = "status_pool",
+        summarizer: Callable[[List[Status], bool], str] = summarize_worst_only,
```

Comment:
The issue here would be if the library  becomes widely used, we'd get divergence between what gets reported in juju status - since everyone will be tempted to customize it differently.  I think I'd prefer we just have one canonical presentation of overall status - unchangeable.